### DISCUSSION TITLE
Are the bootloaders of reMarkable 1 and 2 locked or not?

### DISCUSSION THREAD

**Discussion by korpe1** [Eeems-Org/remarkable.guide#63] [Q&A] [answered]
Are the bootloaders of reMarkable 1 and 2 locked or not?

So, can I e.g. flash Debian to them if I develop drivers for the Linux kernel?

**Comment by Eeems** [comment, marked as answer]
The device gives you full root access. You can `dd` whatever you want to the partitions, as long as it is setup properly for uboot to boot it.

There is a third party Linux distribution for the rM1 already that you could look into for how to do it: http://www.davisr.me/projects/parabola-rm/

I wouldn't really recommend replacing the OS though, instead I would recommend a chroot: https://github.com/Eeems-Org/remarkable-debian-chroot

You completely void the software warranty when doing this though, as you've replaced the provided software completely.

**Reply by korpe1** [reply]
> The device gives you full root access.

I didn't asked that. A bootloader and root aren't related to each other.
An non-locked bootloader allow user run whatever software he want on his hardware and root is just an operating system (Linux, Unix) spesific thing. Root relates to the bootloader only by the fact that with an non-locked bootloader user can modify the operating system and enable root access.

> You can dd whatever you want to the partitions, as long as it is setup properly for uboot to boot it.

So, I assume, the bootloader isn't locked?

> There is a third party Linux distribution for the rM1 already that you could look into for how to do it: http://www.davisr.me/projects/parabola-rm/

Cool :)

> I wouldn't really recommend replacing the OS though, instead I would recommend a chroot: https://github.com/Eeems-Org/remarkable-debian-chroot

Why?

> You completely void the software warranty when doing this though, as you've replaced the provided software completely.

Of course, but you can always flash the original OS back.

**Reply by Eeems** [reply]
> > The device gives you full root access.
> 
> I didn't asked that. A bootloader and root aren't related to each other. An non-locked bootloader allow user run whatever software he want on his hardware and root is just an operating system (Linux, Unix) spesific thing. Root relates to the bootloader only by the fact that with an non-locked bootloader user can modify the operating system and enable root access.
> 
> > You can dd whatever you want to the partitions, as long as it is setup properly for uboot to boot it.
> 
> So, I assume, the bootloader isn't locked?

Since you seemed to need it to be explicitly stated: the bootloader is not locked.

You can find the bootloader source here: https://github.com/reMarkable/uboot

> 
> > I wouldn't really recommend replacing the OS though, instead I would recommend a chroot: https://github.com/Eeems-Org/remarkable-debian-chroot
> 
> Why?

- Proprietary drivers for wifi and the screen that is easier to get updates for by just running the official OS.
- Lack of drivers at all for the screen on the rM2.
- Access to third party software developed specifically for the device on the official OS.

> 
> > You completely void the software warranty when doing this though, as you've replaced the provided software completely.
> 
> Of course, but you can always flash the original OS back.

Assuming you haven't bricked the device, sure. But if you brick the device it won't be covered.


---

Side-note, I'm a volunteer trying to answer questions here. Being snarky and talking down to me isn't helpful if you want to continue to get responses.

**Reply by korpe1** [reply]
> Side-note, I'm a volunteer trying to answer questions here. Being snarky and talking down to me isn't helpful if you want to continue to get responses.

Oh, sorry about that, really. That wasn't my intention. I wanted just point that fact because so many messing with bootloaders and root access.

Thank you for quick answers and your work overall around reMarkable!
